### DISCUSSION TITLE
Consider CJS or help with ESM (for Electron, nw, proton-native)

### DISCUSSION THREAD

**Discussion by pastinepolenta** [webtorrent/webtorrent#2397] [Q&A]
I have noticed that now `webtorrent-hybrid` is ESM only. 
While I probably can see the reasons of this choice I am struggling using webtorrent in any of the popular cross platform frameworks. I tried:

- nw
- electron
- proton native

None of them seems to support ESM. 

Thus, I am using an old version of webtorrent (4.x). You could argue that is on *them* to add ESM support, but it doesn't seem to be a trivial task and  it's not gonna happen any time soon.

- Maybe there are some workarounds, libraries etc to fix this?
- Maybe you can suggest some ways?
- Maybe you can bring back CJS support?
- ~~How to you package the webtorrent-desktop app anyways?~~ (EDIT: I see it uses webtorrent, not hybrid)

Thanks!

**Comment by DiegoRBaquero** [comment]
webtorrent 1.x is still CJS :)

There's no webtorrent 4.x ??

**Reply by pastinepolenta** [reply]
sorry I will edit my question, I am referring to webtorrent-hybrid as it allows both WebRTC and TCP peers 
(the hybrid repo does not have discussions enabled so I wrote here)

**Reply by DiegoRBaquero** [reply]
If you are using electron, you do not need to use webtorrent-hybrid. You get access to both type of peers just like WebTorrent Desktop.

And once ESM is in place, we won't support CJS anymore

**Reply by pastinepolenta** [reply]
Maybe you can point me to some documentation that clarifies how to achieve this?

as per readme webtorrent on node can access TCP peers, while on the client can access WebRTC peers and we need hybrid to have both. If this is not the case it should be stated somewhere?

Also, I was actually using NW cause I was getting better bundle sizes (might switch back to Electron if needed). I assume there is no way to achieve this with NW?

**Reply by DiegoRBaquero** [reply]
On electron you'd need to run `webtorrent` on the main process, not "painter" iirc.

And NW, no idea. Never used it
